### PR TITLE
feat: add structured metadata to Diagnostic model

### LINKED ISSUE #353: Add structured metadata to Diagnostic model
## Context
From code audit (architecture review). The `Diagnostic` struct lacks structured metadata for tooling integration.

## Current State
- Rule ID is an unvalidated string
- No category/group info for filtering
- Fixes are untyped (no distinction between insertion, deletion, config update)
- Users can't filter diagnostics by tool/category without string parsing

## Proposal
1. Add optional `rule_metadata: Option<RuleMetadata>` with category, severity, applies_to
2. Validate rule IDs at creation time using `agnix-rules`
3. Add `Fix` subtypes for better IDE integration
4. Include metadata in JSON/SARIF output

## Location
`crates/agnix-core/src/diagnostics.rs:80-101`

## Effort
Low - mostly additive with optional fields

### PR DESCRIPTION
**Pull request by avifenesh**
## Summary
- Add `RuleMetadata` struct (category, severity, applies_to_tool) to the `Diagnostic` model
- Auto-populate metadata at construction time via build-time lookup table generated from `rules.json`
- Surface metadata in all output formats: JSON, SARIF, MCP, LSP, and CLI (verbose)
- Backward-compatible: uses `Option<RuleMetadata>` with serde skip_serializing_if

## Changes
- `agnix-rules`: Extend build.rs to generate `RULES_METADATA` constant; add `get_rule_metadata()` API
- `agnix-core`: Add `RuleMetadata` struct and optional `metadata` field to `Diagnostic`; auto-populate in constructors
- `agnix-cli`: Surface metadata in JSON, SARIF, and verbose text output
- `agnix-lsp`: Include metadata in diagnostic data payload with backward-compatible deserialization
- `agnix-mcp`: Add metadata fields to `DiagnosticOutput`

## Test Plan
- [x] 30+ new tests across all crates (auto-populate, serde roundtrip, output formats, edge cases)
- [x] Validator integration test verifying real validators produce metadata
- [x] Backward compatibility tests for LSP data format and serde deserialization
- [x] All 2441+ existing tests pass
- [x] cargo clippy clean (zero warnings)

Closes #353

### CODE REVIEW

File: crates/agnix-cli/src/sarif.rs [hunk 1]
```diff
@@ -114,15 +131,39 @@ static RULES: LazyLock<Vec<ReportingDescriptor>> = LazyLock::new(|| {
...
+                        severity: if severity.is_empty() {
+                            None
+                        } else {
+                            Some(severity.to_string())
+                        },
+                        applies_to_tool: if tool.is_empty() {
+                            None
+                        } else {
+                            Some(tool.to_string())
+                        },
```
[the remark this answers is not in the document]
Comment (avifenesh) [reply crates/agnix-cli/src/sarif.rs]: Addressed in commit 966abb3. Refactored to use the idiomatic .then() pattern as suggested.

File: crates/agnix-core/src/diagnostics.rs [hunk 2]
```diff
@@ -107,6 +131,19 @@ pub enum DiagnosticLevel {
...
+/// Build a `RuleMetadata` from the compile-time rules catalog.
+fn lookup_rule_metadata(rule_id: &str) -> Option<RuleMetadata> {
+    agnix_rules::get_rule_metadata(rule_id).map(|(category, severity, tool)| RuleMetadata {
+        category: category.to_string(),
+        severity: severity.to_string(),
+        applies_to_tool: if tool.is_empty() {
+            None
+        } else {
+            Some(tool.to_string())
+        },
```
[the remark this answers is not in the document]
Comment (avifenesh) [reply crates/agnix-core/src/diagnostics.rs]: Addressed in commit 966abb3. Refactored to use the idiomatic .then() pattern as suggested.
